### PR TITLE
Drop old SF code

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is BC.

### Subject

Removing old code that never gets executed because we dropped SF 2.3 to 2.7

### CODE REVIEW

File: src/DependencyInjection/SonataCoreExtension.php [hunk 1]
```diff
@@ -196,11 +196,9 @@ public function configureSerializerFormats($config)
     protected function deprecateSlugify(ContainerBuilder $container)
     {
         $definition = $container->getDefinition('sonata.core.slugify.cocur');
-        if (method_exists($definition, 'setDeprecated')) {
-            $definition->setDeprecated(true);
+        $definition->setDeprecated(true);
```
Comment (greg0ire) [inline src/DependencyInjection/SonataCoreExtension.php]: Can be simplified further: `$container->getDefinition('sonata.core.slugify.cocur')->setDeprecated(true);`, same below

File: src/Form/Type/DateTimeRangeType.php [hunk 2]
```diff
@@ -109,11 +110,7 @@ public function configureOptions(OptionsResolver $resolver)
             'field_options' => [],
             'field_options_start' => [],
             'field_options_end' => [],
-            // NEXT_MAJOR: Remove ternary and keep 'Symfony\Component\Form\Extension\Core\Type\DateTimeType'
-            // (when requirement of Symfony is >= 2.8)
-            'field_type' => method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
-                ? 'Symfony\Component\Form\Extension\Core\Type\DateTimeType'
-                : 'datetime',
+            DateTimeType::class,
```
Comment (greg0ire) [inline src/Form/Type/DateTimeRangeType.php]: You forgot `'field_type'` here

**Comment by jordisala1991** [comment]
fixed both

**Comment by jordisala1991** [comment]
Note: squash this PR on merge

**Comment by jordisala1991** [comment]
Everything squashed, can you review again @greg0ire ?

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

**Pull request merged**